datasets/wiki_dpr/wiki_dpr.py

```diff
@@ -29,42 +29,71 @@
 
 _VECTORS_URL = "https://dl.fbaipublicfiles.com/dpr/data/wiki_encoded/single/nq/wiki_passages_{i}"
 
+_INDEX_URL = "https://storage.googleapis.com/huggingface-nlp/datasets/wiki_dpr"
+
 
 class WikiDprConfig(nlp.BuilderConfig):
     """BuilderConfig for WikiDprConfig."""
 
-    def __init__(self, with_embeddings=True, with_index=False, index_train_size=262144, dummy=False, **kwargs):
+    def __init__(
+        self,
+        with_embeddings=True,
+        with_index=False,
+        index_train_size=262144,
+        dummy=False,
+        index_type="compressed",
```

Comment:
So maybe we call this `index_name` and:
- store the index with a simple name like `compressed` and `exact`
- add metadata to the index somewhere in an additional `index_name.json` file or in the `dataset.json` (sorry I know it's one additional file but I feel like we might need a better way to store index metadata than the file name in the future)

Reply:
what do you think?

Reply:
I was thinking of adding the post processing resources (like the indexes) metadata (resource name, resource file name, size) in the DatasetInfo.

However `wiki_dpr` may have an index or not depending on the configuration (`with_index` parameter) so it's not ideal to do that.

More precisely:

A `dataset_info.json` is inside cache directory. A cache directory corresponds to a "downloaded and prepared" dataset  for a given configuration name + version + script hash. It means that for one cache dir we could have one dataset with the exact index, a one with the compressed index, but only one `dataset_info.json` and that's probably not what we want.

One solution to fix that would be to take the `config_kwargs` into account when building the `_relative_cache_dir`.
Another solution would to have one configuration name for exact index, one for compressed index, and one for no index.

Reply:
I'm changing wiki_dpr to have one configuration name for all the combinations of
- dummy (true/false)
- wiki split (only one possibility for now: psgs_w100)
- embeddings name (no embeddings or "nq")
- index name (no index or "compressed" or "exact")

Then we'll have one set of post processing resources per config and we'll be fine to add indexes metadata in DatasetInfo.